configuration/configurator/ExceptionStrings.hpp

```diff
@@ -0,0 +1,104 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef _EXCEPTION_STRINGS_HPP_
+#define _EXCEPTION_STRINGS_HPP_
+
+#include "jexcept.hpp"
+#include <cstring>
+
+enum eExceptionCodes
+{
+    EX_STR_CAN_NOT_OPEN_XSD  = 1,
+    EX_STR_SIMPLE_TYPE_ALREADY_DEFINED,
+    EX_STR_COMPLEX_TYPE_ALREADY_DEFINED,
+    EX_STR_ATTRIBUTE_GROUP_ALREADY_DEFINED,
+    EX_STR_CAN_NOT_PROCESS_ENV_XML,
+    EX_STR_XPATH_DOES_NOT_EXIST_IN_TREE,
+    EX_STR_MISSING_REQUIRED_ATTRIBUTE,
+    EX_STR_MISSING_VALUE_ATTRIBUTE_IN_LENGTH,
+    EX_STR_LENGTH_VALUE_MUST_BE_GREATER_THAN_OR_EQUAL_TO_ZERO,
+    EX_STR_PATTERN_HAS_INVALID_VALUE,
+    EX_STR_WHITE_SPACE_HAS_INVALID_VALUE,
+    EX_STR_MISSING_NAME_ATTRIBUTE,
+    EX_STR_UNKNOWN,
+    EX_STR_LAST_ENTRY
+};
+
+namespace CONFIGURATOR
+{
+#define MAX_EXCEPTION_STRING_LENGTH 2048
+
+#define CATCH_EXCEPTION_AND_EXIT \
+catch (::IException *except) \
+{\
+    StringBuffer strErrMsg;\
+    except->errorMessage(strErrMsg);\
+    ::std::cout << ::std::endl << strErrMsg.str() << ::std::endl << ::std::endl;\
+    exit(-1);\
```

Comment:
It prob doesn't matter since you're exiting, but typically you'd release the exception before returning.